python/cuml/test/test_train_test_split.py

```diff
@@ -267,3 +265,37 @@ def test_split_array_single_argument(type, test_size, train_size, shuffle):
         X_rec = cp.sort(cp.concatenate(X_train, X_test))
 
         assert X_rec == X
+
+
+@pytest.mark.parametrize('type', test_array_input_types)
+@pytest.mark.parametrize('test_size', [0.2, 0.4, None])
+@pytest.mark.parametrize('train_size', [0.6, 0.8, None])
+def test_stratified_split(type, test_size, train_size):
+    X, y = make_classification()
+
+    if type == 'cupy':
+        X = cp.asarray(X)
+        y = cp.asarray(y)
+
+    if type == 'numba':
+        X = cuda.to_device(X)
+        y = cuda.to_device(y)
+
+    def counts(y):
+        _, y_indices = cp.unique(y, return_inverse=True)
+        class_counts = cp.bincount(y_indices)
+        total = cp.sum(class_counts)
+        percent_counts = []
+        for count in (class_counts):
+            percent_counts.append(cp.around(float(count)/total.item(),
+                                            decimals=2).item())
+        return percent_counts
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                        train_size=train_size,
+                                                        test_size=test_size,
+                                                        stratify=True)
+
+    original_counts = counts(y)
+    split_counts = counts(y_train)
+    np.isclose(original_counts, split_counts, equal_nan=False)
```

Comment:
Is there another check we could do besides the counts to know we're generating the different objects correctly?
This also seems to only test `y_train` and not the rest of the objects

Reply:
It's hard to make comparisons because of the shuffle. I've dtype checks. I can't think of other checks that might make sense here